synapseclient/api/entity_factory.py

```diff
@@ -400,12 +400,19 @@ def _check_entity_restrictions(
     if restriction_information and restriction_information.get(
         "hasUnmetAccessRequirement", None
     ):
-        warning_message = (
-            "\nThis entity has access restrictions. Please visit the web page for this entity "
-            f'(syn.onweb("{synapse_id}")). Look for the "Access" label and the lock icon underneath '
-            'the file name. Click "Request Access", and then review and fulfill the file '
-            "download requirement(s).\n"
-        )
+        if not syn.credentials._token:
```

Comment:
Could we also check for `not syn.credentials`?

Reply:
no, `syn.credentials` is not None in these cases, but `token` (exposed as `_token`) is an empty string.

```
SynapseAuthTokenCredentials(username='anonymous', displayname='Anonymous', token='', owner_id='273950')
```

Reply:
Or did you mean do both checks?

Reply:
added that check and a test in my latest [commit](https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1185/commits/a9d7bd47382d3c3753b8c5bca5514063df202f40)